crates/sql/src/ast/subscription.rs

```diff
@@ -0,0 +1,323 @@
+//! ## Subscriptions AST
+//!
+//! The subscription language is strictly a query language.
+//!
+//! There is no context for manually updating this view.
+//!
+//! > **NOTE:** Because subscriptions are evaluated in realtime,
+//! > performance is critical, and as a result,
+//! > additional restrictions are applied over [`super::ad_hoc`] queries.
+//! > These restrictions are highlighted below.
+//!
+//! - Only `SELECT *` or `table.*` projections
+//! - No `ORDER BY`, `LIMIT`, or aggregates
+//! - No `OR` conditions
+//! - No `!=` predicates
+//! - Only 2-way joins
+//!
+//! ### Statements
+//!
+//! - [`SQL`](SqlAst)
+//!
+//! ### `SELECT`
+//!
+//! - [`SELECT`](SqlProject)
+//! - [`FROM`](SqlFrom)
+//! - [`JOIN`](SqlJoin)
+//! - [`ON`](SqlOn)
+//! - [`WHERE`](SqlSelection)
+//!
+//! ### Subscription Grammar
+//!
+//! **WARNING:** This grammar does not guarantee the semantic validity of its queries, only their syntactic correctness.
+//!
+//! ```ebnf
+//! query
+//!     = SELECT projection FROM relation [ WHERE predicate ]
+//!     ;
+//!
+//! projection
+//!     = STAR
+//!     | ident '.' STAR
+//!     ;
+//!
+//! relation
+//!     = table [ [AS] ident ] [ [INNER] JOIN table [ [AS] ident ] ON predicate_on ]
+//!     ;
+//!
+//! columnExpr
+//!     = column
+//!     | qualfied
+//!     ;
+//!
+//! predicate
+//!     =  expr
+//!     | predicate AND predicate
+//!     ;
+//!
+//! predicate_on
+//!     =  columnExpr '=' columnExpr
+//!     | predicate_on AND predicate_on
+//!     ;
+//!
```

Comment:
This is an addition to the spec, because the `predicate` allows expressions that will not optimize.